module/data/activity/base-activity.mjs

```diff
@@ -146,6 +146,244 @@ export default class BaseActivityData extends foundry.abstract.DataModel {
+    foundry.utils.setProperty(source, "flags.dnd5e.persistSourceMigration", true);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetch activation data from the item source and transform it into an activity's activation object.
+   * @param {object} source  Item's candidate source data to transform.
+   * @returns {object}       Creation data for new activity.
+   */
+  static transformActivationData(source) {
+    return {
+      type: source.system.activation?.type === "none" ? "" : (source.system.activation?.type ?? ""),
+      value: source.system.activation?.cost ?? null,
+      condition: source.system.activation?.condition ?? ""
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetch activation data from the item source and transform it into an activity's consumption object.
+   * @param {object} source  Item's candidate source data to transform.
+   * @returns {object}       Creation data for new activity.
+   */
+  static transformConsumptionData(source) {
+    const targets = [];
+
+    const type = {
+      attribute: "attribute",
+      hitDice: "hitDice",
+      material: "material",
+      charges: "itemUses"
+    }[source.system.consume?.type];
+
+    if ( type ) targets.push({
+      type,
+      target: source.system.consume?.target ?? "",
+      value: source.system.consume?.amount ?? "1",
+      scaling: {
+        mode: source.system.consume?.scale ? "amount" : "",
+        formula: ""
+      }
+    });
+
+    // If no target type set but this item has max uses, set consumption type to itemUses with blank target
+    else if ( source.system.uses?.max ) targets.push({
+      type: "itemUses",
+      target: "",
+      value: "1",
+      scaling: {
+        mode: source.system.consume?.scale ? "amount" : "",
+        formula: ""
+      }
+    });
+
+    return {
+      targets,
+      scaling: {
+        allowed: source.system.consume?.scale ?? false,
+        max: ""
+      }
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetch action data from the item source and transform it into an activity's damage parts array.
+   * @param {object} source  Item's candidate source data to transform.
+   * @param {string[]} part  The damage part to transform.
+   * @returns {object}       Creation data for new activity.
+   */
+  static transformDamagePartData(source, [formula, type]) {
+    const data = {
+      number: null,
+      denomination: null,
+      bonus: "",
+      types: type ? [type] : [],
+      custom: {
+        enabled: false,
+        formula: ""
+      },
+      scaling: {
+        mode: source?.system.scaling?.mode !== "none" ? "whole" : "",
+        number: null,
+        formula: source?.system.scaling?.formula ?? ""
+      }
+    };
+
+    const parsed = (formula ?? "").match(/^\s*(\d+)d(\d+)(?:\s*([+|-])\s*(@?[\w\d.]+))?\s*$/i);
```

Comment:
Could probably simplify the regex by removing spaces from the formula first, e.g. `(formula ?? "").replaceAll(" ", "").match()`. Even so, how confident are we that this parsing is going to catch what we want and not accidentally drop any data? It seems like it would be safer to just migrate legacy damage parts straight to formula.

Reply:
Can't remove spaces since `@steve-1` is not the same as `@steve - 1`.

Reply:
The regex should be fairly narrow so it should fall back to custom formula for anything but the expected case:

```
1d4 // parsed
1d4 + @mod // parsed
1d4 - 5 // parsed
1d3 // custom
5 // custom
1d4 + @mod + 1 // custom
```

I don't think there is any chance of losing data because of the limited scope of the RegEx.

Reply:
There are also a ton of users that will have damage type annotations in the formulas, e.g. `1d6 + @mod[fire]`, looks like those will also be punted to `custom`, but figured it was worth a mention